aqt/installer.py

```diff
@@ -632,6 +640,43 @@ def _warn_on_deprecated_parameter(self, parameter_name: str, value: str):
             f"In the future, please omit this parameter."
         )
 
+    @staticmethod
+    def _get_missing_desktop_arch(host: str, target: str, version: Version, base_dir: Path) -> Optional[str]:
+        """
+        For mobile Qt installations, the desktop version of Qt is a dependency.
+        If the desktop version is not installed, this function returns the architecture that should be installed.
+        If no desktop Qt is required, or it is already installed, this function returns None.
+        """
+        if target not in ["ios", "android"]:
+            return None
+        if host != "windows":
+            arch = aqt.updater.default_desktop_arch_dir(host, version)
+            expected_qmake = base_dir / format(version) / arch / "bin/qmake"
+            return arch if not expected_qmake.is_file() else None
+        else:
+            existing_desktop_qt = QtRepoProperty.find_installed_qt_mingw_dir(base_dir / format(version))
+            if existing_desktop_qt:
+                return None
+            return MetadataFactory(ArchiveId("qt", host, "desktop")).fetch_default_desktop_arch(version)
+
+    def _handle_missing_desktop_qt(self, host: str, target: str, version: Version, base_dir: Path, should_warn: bool):
+        missing_desktop_arch = Cli._get_missing_desktop_arch(host, target, version, base_dir)
+        if not missing_desktop_arch:
+            return
+
+        msg_prefix = (
+            f"You are installing the {target} version of Qt, which requires that the desktop version of Qt "
+            f"is also installed."
+        )
+        if should_warn:
+            self.logger.warning(
+                f"{msg_prefix} You can install it with the following command:\n"
+                f"          `aqt install-qt {host} desktop {version} {missing_desktop_arch}`"
+            )
+        else:
+            self.logger.info(f"{msg_prefix} Now installing Qt: desktop {version} {missing_desktop_arch}")
+            self.run(["install-qt", host, "desktop", format(version), missing_desktop_arch])
```

Comment:
I am thinking about changing this to:
```suggestion
            self.run(["install-qt", host, "desktop", format(version), missing_desktop_arch, "--archives", "qtbase", "qtdeclarative"])
```
... to save on network traffic, because that's what our current CI script does.

I'm not sure if this is the best idea though. I'm not sure which archives are actually required for android/ios, and I don't want our users to be left with an installation that doesn't work.

Reply:
The Azure Pipelines jobs don't appear to run any slower than they did before.